tensornetwork/contractors/opt_einsum_paths/path_contractors.py

```diff
@@ -129,20 +189,30 @@ def auto(net: network.TensorNetwork,
   if n == 1:
     edges = net.get_all_nondangling()
     net.contract_parallel(edges.pop())
+    final_node = list(net.nodes_set)
+    if (len(final_node[0].edges) <= 1) and (output_edge_order is None):
+      output_edge_order = list((net.get_all_edges() -
+                                net.get_all_nondangling()))
+    elif (len(final_node[0].edges) > 1) and (output_edge_order is None):
+      raise ValueError("if the final node has more than one dangling edge"
+                       ", `output_edge_order` has to be provided")
+
+    net.nodes_set = set([list(net.nodes_set)[0].reorder_edges(
```

Comment:
I'd rather you not access `nodes_set` directly. Can you just do `net.get_final_node().reorder_edges(...)`?